### DISCUSSION TITLE
How to follow Getx-Pattern with get-cli and StateMixin elegantly.

### DISCUSSION THREAD

**Discussion by fengwensheng** [jonataslaw/getx#1057] [Q&A]
If one controller has more than one provider, how to deliver the ‘data-s’ to the UI with StateMixin.Here’s my code snippet.
```
class IndexController extends GetxController with StateMixin<List<Data>> {
  final _bannersProvider = Get.find<BannersProvider>();
  final _topProvider = Get.find<TopProvider>(); //things goes well until i want to add another provider😂
  @override
  void onInit() {
    super.onInit();
    _bannersProvider.getBanners().then((resp) {
      change(resp.body.data, status: RxStatus.success());
    }, onError: (e) {
      change(null, status: RxStatus.error());
    });
  }
}
```
If this is not the proper use case for StateMixin, and what is the best practice for it, and I use the get-cli and getx-pattarn to build my project structure.  
I’m newly to GetX and looking forward to your kindly help.

**Comment by ryanyulianto** [comment]
have you found a solution? I also have a problem like that, maybe you can provide a solution if you have found a solution
